surround/tests/cli/remote_cli/add_test.py

```diff
@@ -3,37 +3,52 @@
 import unittest
 import subprocess
 
+from surround.remote.local import Local
+
 __author__ = 'Akshat Bajaj'
 __date__ = '2019/03/04'
 
 class AddTest(unittest.TestCase):
 
     def test_rejecting_path(self):
         process = subprocess.run(['surround', 'init', './', '-p', 'temp', '-d', 'temp', '-w', 'no'], encoding='utf-8', stdout=subprocess.PIPE)
-        self.assertEqual(process.stdout, "info: project created at " + os.getcwd() + "/temp\n")
+        self.assertIn("info: project created at", process.stdout)
+        self.assertIn("temp", process.stdout)
 
         is_temp = os.path.isdir(os.path.join(os.getcwd() + "/temp"))
         self.assertEqual(is_temp, True)
 
-        process = subprocess.run(['surround', 'add', 'data', 'temp.jpg'], encoding='utf-8', stdout=subprocess.PIPE, cwd='temp')
-        self.assertEqual(process.stdout, "error: no remote named data\n")
+        local_remote = Local()
+
+        os.chdir("temp")
+        result = local_remote.add('data', 'temp.jpg')
+        os.chdir("../")
+        self.assertEqual(result, "error: no remote named data")
 
         process = subprocess.run(['surround', 'remote', '-a', '-n', 'data', '-u', os.getcwd(), '-t', 'data'], encoding='utf-8', stdout=subprocess.PIPE, cwd='temp')
 
-        process = subprocess.run(['surround', 'add', 'data', 'temp.jpg'], encoding='utf-8', stdout=subprocess.PIPE, cwd='temp')
-        self.assertEqual(process.stdout, "error: temp.jpg not found.\n")
+        os.chdir("temp")
+        result = local_remote.add('data', 'temp.jpg')
+        os.chdir("../")
+        self.assertEqual(result, "error: temp.jpg not found.")
 
     def test_happy_path(self):
         process = subprocess.run(['surround', 'init', os.getcwd(), '-p', 'temp', '-d', 'temp', '-w', 'no'], encoding='utf-8', stdout=subprocess.PIPE)
-        self.assertEqual(process.stdout, "info: project created at " + os.getcwd() + "/temp\n")
+        self.assertIn("info: project created at", process.stdout)
+        self.assertIn("temp", process.stdout)
 
         is_temp = os.path.isdir(os.path.join(os.getcwd() + "/temp"))
         self.assertEqual(is_temp, True)
 
         process = subprocess.run(['surround', 'remote', '-a', '-n', 'data', '-u', os.getcwd(), '-t', 'data'], encoding='utf-8', stdout=subprocess.PIPE, cwd='temp')
 
-        process = subprocess.run(['surround', 'add', 'data', os.getcwd() + '/surround/remote/base.py'], encoding='utf-8', stdout=subprocess.PIPE, cwd="temp")
-        self.assertEqual(process.stdout, "info: file added successfully\n")
+        local_remote = Local()
+
+        os.chdir("temp")
+        result = local_remote.add('data', os.path.join(os.getcwd(), '..', 'surround/remote/base.py'))
```

Comment:
What's the use of `..`?

Reply:
To go back to the root of the repository so we can get the path to the file we're uploading as a test